### ISSUE TITLE
modules can safely import libraries of the same name.

### ISSUE THREAD

**Issue by osynge** [saltstack/salt#35444]
### Description of Issue/Question

The namespace of modules is not mangled, hence importing a library of the same name, can cause a false positive, and so can cause issues, such as infinite recursion.

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)

create and external module called elephant.

eg _modules/elephant/__init__.py

with content:

# Import Python Libs
from __future__ import absolute_import
import logging

log = logging.getLogger(__name__)

__virtualname__ = 'elephant'

try:
    import elephant
    HAS_ELEPHANT = True
except ImportError:
    HAS_ELEPHANT = False

def __virtual__():
    if HAS_ELEPHANT is False:
        msg = 'elephant unavailable: {0} execution module cant be loaded '.format(__virtualname__)
        return False, msg
    return __virtualname__

def test_elephant():
    return elephant.test_elephant()

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

Do the above and then 

    sync.all

then call the test method. 

salt '*' elepehant.test_elephant

You should get the exception:

     RuntimeError: maximum recursion depth exceeded

Where as the correct result woudl have been to fail to import elephant.

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)

Salt Version:
           Salt: 2015.8.10
 
Dependency Versions:
         Jinja2: 2.8
       M2Crypto: Not Installed
           Mako: 1.0.1
         PyYAML: 3.11
          PyZMQ: 15.2.0
         Python: 2.7.12 (default, Jul 01 2016, 15:34:22) [GCC]
           RAET: Not Installed
        Tornado: 4.3
            ZMQ: 4.1.5
           cffi: 1.6.0
       cherrypy: 3.6.0
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
        libgit2: 0.22.0
        libnacl: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: 1.2.5
      pycparser: 2.14
       pycrypto: 2.6.1
         pygit2: 0.22.0
   python-gnupg: 0.3.8
          smmap: Not Installed
        timelib: Not Installed
 
System Versions:
           dist: SuSE 42.1 x86_64
        machine: x86_64
        release: 4.1.27-27-default
         system: openSUSE  42.1 x86_64

**Comment by Ch3LL** [comment]
@oms4suse agreed we could be spitting out a better error here instead of the recursive error that is more useful for the user to know what the actual cause is. Thanks.
